File: clients/rust/src/models/cache_set_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize)]
pub struct CacheSetIn {}

impl CacheSetIn {
    pub fn new() -> Self {
        Self {}
    }
}

#[derive(Clone, Debug, Deserialize, Serialize)]
pub(crate) struct CacheSetIn_ {
    pub key: String,

    pub value: Vec<u8>,

    /// Time to live in milliseconds
    pub ttl: u64,
}

File: clients/javascript/src/models/cacheSetIn.ts
// this file is @generated

export interface CacheSetIn {
    value: number[];
    /** Time to live in milliseconds */
    ttl: number;
}

export interface CacheSetIn_ {
    key: string;
    value: number[];
    /** Time to live in milliseconds */
    ttl: number;
}

export const CacheSetInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): CacheSetIn_ {
        return {
            key: object['key'],
            value: object['value'],
            ttl: object['ttl'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: CacheSetIn_): any {
        return {
            'key': self.key,
            'value': self.value,
            'ttl': self.ttl,
        };
    }
}